### PR TITLE
Create bus details

### PR DESCRIPTION
**Pull request by priyanshithapa23**
Creating bus details by admin

### CODE REVIEW

File: app.js
Comment (khushbukumari15) [inline app.js:1]: repetition  of  require routes

File: controllers/bus.controller.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const busesService=require('../services/bus.services')
+
+const busDetails=async function (req,res){
+    const insertingDetails=req.body
+    const addBusData=await busesService.validation(insertingDetails)
```
Comment (mit456) [inline controllers/bus.controller.js]: Function name can be better, only `validation` does not make much sense.

File: services/bus.services.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return responseConst.missingFieldValidationError 
+    }
+    if (bodyData.totalSeats < 0 && bodyData.totalSeats <=40){
+        return responseConst.seatRangeValidationError
+    }
+    if (bodyData.days.length == 0 || !Array.isArray(bodyData.days) ){
+        return responseConst.seatValueTypeValidationError
+    }
+         
+    if (!bodyData.days=="Monday" || !bodyData.days == "Tuesday" || !bodyData.days == "Wednesday" || !bodyData.days == "Thursday" || !bodyData.days == "Friday" || !bodyData.days == "Saturday" || !bodyData.days == "Sunday" ){
```
Comment (mit456) [inline services/bus.services.js]: We can approach this problem in a better way. Lets discuss over standup.

File: Models/bus.model.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+async function insertBusDetails(doc){
+    const client = new MongoClient(dbConstants.uri);
+    try{
+        const database=client.db(dbConstants.dbName);
+        const collection=database.collection(dbConstants.busCollection)
+
+        const result = await collection.insertOne(doc);
+        resBody={
+            "message": "successfully inserted"
```
Comment (mit456) [inline Models/bus.model.js]: This should be part of response constants file.

File: constants/db.constants.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const dbConst = {
+    dbName:"busBooking",
+    busCollection:"Buses",
```
Comment (mit456) [inline constants/db.constants.js]: make it small case.

Comment (mit456) [reply constants/db.constants.js]: `buses`

File: controllers/bus.controller.js (hunk 1 above)
Comment (mit456) [reply controllers/bus.controller.js]: For e.g `validateBusDetails`

File: services/bus.services.js (hunk 2 above)
Comment (mit456) [reply services/bus.services.js]: Discussed, @khushbukumari15 will work on it.

**Review by mit456 — APPROVED** [review]

**Pull request merged**